hearth/spec/hearth/telemetry/otel_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'opentelemetry-sdk'
+
+module Hearth
+  module Telemetry
+    describe OTelProvider do
+      before do
+        allow(Hearth::Telemetry)
+          .to receive(:otel_loaded?)
+          .and_return(true)
+      end
+
+      let(:otel_provider) { OTelProvider.new }
+      let(:context_manager) { otel_provider.context_manager }
+      let(:tracer_provider) { otel_provider.tracer_provider }
+      let(:tracer) { tracer_provider.tracer('some_tracer') }
+
+      describe '#initialize' do
+        it 'raises ArgumentError when otel dependency fails to load' do
+          allow(Hearth::Telemetry).to receive(:otel_loaded?).and_return(false)
+          expect { otel_provider }.to raise_error(ArgumentError)
+        end
+
+        it 'sets up tracer provider and context manager' do
+          expect(tracer_provider).to be_a(OpenTelemetry::Trace::TracerProvider)
+          expect(context_manager).to be_a(Hearth::Telemetry::OTelContextManager)
+        end
+      end
+
+      describe ':context_manager' do
```

Comment:
Should this be ContextManager (the class name) instead of a symbolized method name?